timesketch/api/v1/resources/sigma.py

```diff
@@ -87,15 +125,274 @@ def get(self, rule_uuid):
+                "Missing values from the request.",
+            )
+
+        try:
+            sigma_rule = ts_sigma_lib.parse_sigma_rule_by_text(content)
+
+        except ValueError:
+            logger.error(
+                "Sigma Parsing error with the user provided rule",
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Error unable to parse the provided Sigma rule",
+            )
+
+        except NotImplementedError as exception:
+            logger.error(
+                "Sigma Parsing error: Feature in the rule provided "
+                " is not implemented in this backend",
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Sigma Parsing error: Feature in the rule provided "
+                " is not implemented in this backend {0!s}".format(exception),
+            )
+
+        except sigma_exceptions.SigmaParseError as exception:
+            logger.error("Sigma Parsing error: unknown error", exc_info=True)
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Sigma parsing error generating rule  with error: {0!s}".format(
+                    exception
+                ),
+            )
+
+        except yaml.parser.ParserError as exception:
+            logger.error(
+                "Sigma Parsing error: an invalid yml file has been provided",
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Sigma parsing error: invalid yaml provided {0!s}".format(
+                    exception
+                ),
+            )
+
+        if not sigma_rule:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sigma was parsed")
+
+        return jsonify({"objects": [sigma_rule], "meta": {}})
+
+
+class SigmaRuleListResource(resources.ResourceMixin, Resource):
+    """Resource to get list of SigmaRules."""
+
+    @login_required
+    def get(self):
+        """Handles GET request to the resource.
+
+        Returns:
+            Dict of sigma rules
+        """
+        sigma_rules = []
+
+        all_sigma_rules = SigmaRule.query.all()
+        # Return a subset of the Sigma objects to reduce the amount of
+        # data sent to the client.
+        for rule in all_sigma_rules:
+            sigma_rules.append(_enrich_sigma_rule_object(rule=rule))
+
+        meta = {"rules_count": len(sigma_rules)}
+        return jsonify({"objects": sigma_rules, "meta": meta})
+
+
+class SigmaRuleResource(resources.ResourceMixin, Resource):
+    """Resource to get / delete / post / put a Sigma rule."""
+
+    @login_required
+    def get(self, rule_uuid):
+        """Handles GET request to the Sigma Rule resource.
+
+        Handels GET API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
```

Comment:
I think we're looking for a docstring describing **the effect** of the query rather than the process, e.g. "Fetches a single SigmaRule from the database by filtering on rule_uuid and returns a JSON representation of it"